### ISSUE TITLE
Update Testing Matrix

### ISSUE THREAD

**Issue by pyrodie18** [ansible-collections/community.zabbix#777]
##### SUMMARY
Currently, we only test against the latest version of Zabbix as well as the latest operating systems but report that we support 4 versions of Zabbix and a few older operating systems.  Should update the testing matrix so that it tests all supported versions and operating systems.

##### ISSUE TYPE
- Feature Idea

**Comment by pyrodie18** [comment]
My suspicion is that either Github gives free minutes to large open-source projects, or Ansible (RedHat) is paying for the runners.  Anyone that we could ask?

The other option is that we build a separate trigger event that would do a complete regression test of everything that would be triggered as part of a release event (not sure what the trigger would be exactly but I'm sure we could figure something out).

**Comment by D3DeFi** [comment]
@Andersson007 would you be able to shine some light into this matter? :)

**Comment by pyrodie18** [comment]
> @Andersson007 would you be able to shine some light into this matter? :)
> 
> Mostly talking about roles here. We do test plugins against zabbix 3.0, 4.0, 5.0, 6.0, 6.2 with stable-2.10-13 and develop
> 
> side note: should add stable-2.14 probably, right?

ya.  And remove 3.0.  Did we decide if we're going to keep or ditch 4.0 when we role versions?

**Comment by nitzmahone** [comment]
Yes, Ansible's paying for the VMs in a gigantic Azure VMSS that serves both ansible-core and the community collections projects, among other things.

**Comment by webknjaz** [comment]
> Yes, Ansible (well, Red Hat) is paying for the VMs in a gigantic Azure VMSS that serves both ansible-core and the community collections projects, among other things.

This project doesn't seem to make use of the Ansible Core CI, though. Evidently, it uses the free minutes that GH has for public projects: https://github.com/ansible-collections/community.zabbix/actions/runs/2867704689/usage.

P.S. By the way, @pyrodie18 if you're looking for areas of improvements, try integrating https://github.com/marketplace/actions/ansible-test — it helps reduce some boilerplate the collection maintainers end up supporting and is already being offered by the collection template project by default: https://github.com/ansible-collections/collection_template/blob/main/.github/workflows/ansible-test.yml.

**Comment by D3DeFi** [comment]
Thanks for clarifying @nitzmahone @webknjaz ! Yes, I don't think that we use Azure in any way, we also cannot utilize ansible-test container as we do need to test same codebase against multiple versions of zabbix-server.

Is there a way for us to see how far we are away from Github's free 2000 CI/CD minutes?

**Comment by pyrodie18** [comment]
According to the documentation public projects aren't charged.  Just like if we were to do a self-hosted instance.  That being the case I think we're good to go.

**Comment by webknjaz** [comment]
> as we do need to test same codebase against multiple versions of zabbix-server.

You could use `--docker-network` to make zabbix containers accessible :man_shrugging:

**Comment by D3DeFi** [comment]
maybe this would be worth investigating. I don't remember why we decided to create our own CI logic in the past, but I know there were some complications that prevented us from being productive when we used `ansible-test --docker`
